app/web_conn.go

```diff
@@ -732,7 +732,8 @@ func (wc *WebConn) shouldSendEvent(msg *model.WebSocketEvent) bool {
 			hasReadPrivateDataPermission = model.NewBool(wc.App.RolesGrantPermission(wc.GetSession().GetUserRoles(), model.PermissionManageSystem.Id))
 		}
 
-		if !*hasReadPrivateDataPermission {
+		// sensitive data should be allowed to be emitted back to source-user event.
+		if !*hasReadPrivateDataPermission && (wc.UserId != msg.GetBroadcast().UserId) {
```

Comment:
We can revert this change. We only need to set `ContainsSanitizedData`, when we need to differentiate between users and admins. It's not perfectly made, but `ContainsSensitiveData` goes together with `ContainsSanitizedData` to separate users and admins.

Since we are directly sending the message to the user, we can ignore this part and it will match the condition in L742.

Reply:
Understood the context, thanks @agnivade for the explanation. I've reverted these changes. Please review again?